File: backend/src/main/java/com/nlu/WebThuongMai/dto/response/productResp/ProductResponse.java
package com.nlu.WebThuongMai.dto.response.productResp;

import com.nlu.WebThuongMai.dto.response.promotion.PromotionResponse;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.Set;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ProductResponse {
    long id;
    String name;
    double price;
    String status;
    Set<CategoryResponse> categories;
    String description;
    Set<String> images;
    int hot;
    Set<PromotionResponse> promotions;
}

File: frontend/src/app/core/models/response/product-response/productResponse.ts
import { CategoryResponse } from "./CategoryResponse";
import { PromotionResponse } from "./PromotionResponse";

export interface ProductResponse {
  id: number;
  name: string;
  price: string;
  status: string;
  categories: CategoryResponse[];
  description: string;
  images: string[];
  hot: number;
  promotions: PromotionResponse[];
}

    